modules/gapi/samples/onevpl_infer_single_roi.cpp

```diff
@@ -0,0 +1,254 @@
+
+    // get file name
+    std::string file_path = cmd.get<std::string>("input");
+    const std::string output = cmd.get<std::string>("output");
+    const auto face_model_path = cmd.get<std::string>("facem");
+
+    // check ouput file extension
+    if (!output.empty()) {
+        auto ext = output.find_last_of(".");
+        if (ext == std::string::npos || (output.substr(ext + 1) != "avi")) {
+            std::cerr << "Output file should have *.avi extension for output video" << std::endl;
+            return -1;
+        }
+    }
+
+    auto face_net = cv::gapi::ie::Params<custom::FaceDetector> {
+        face_model_path,                 // path to topology IR
+        get_weights_path(face_model_path),   // path to weights
+        cmd.get<std::string>("faced"),   // device specifier
+    };
+    auto kernels = cv::gapi::kernels
+        < custom::OCVLocateROI
+        , custom::OCVParseSSD
+        , custom::OCVBBoxes>();
+    auto networks = cv::gapi::networks(face_net);
+
+    // Create source
+    cv::Ptr<cv::gapi::wip::IStreamSource> cap;
+    try {
+        cap = cv::gapi::wip::make_vpl_src(file_path);
+        std::cout << "oneVPL source desription: " << cap->descr_of() << std::endl;
+    } catch (const std::exception& ex) {
+        std::cerr << "Cannot create source: " << ex.what() << std::endl;
+        return -1;
+    }
+
+    cv::GMetaArg descr = cap->descr_of();
+    auto frame_descr = cv::util::get<cv::GFrameDesc>(descr);
+    (void)frame_descr;
+
+    // Now build the graph
+    cv::GFrame in;
+    auto size = cv::gapi::streaming::size(in);
+    auto roi = custom::LocateROI::on(size);
+    auto blob = cv::gapi::infer<custom::FaceDetector>(roi, in);
+    auto rcs = custom::ParseSSD::on(blob, roi, size);
+    auto out_frame = cv::gapi::wip::draw::renderFrame(in, custom::BBoxes::on(rcs, roi));
+    auto out = cv::gapi::streaming::BGR(out_frame);
+
+    cv::GStreamingCompiled pipeline;
+    try {
+        pipeline  = cv::GComputation(cv::GIn(in), cv::GOut(out))
+                .compileStreaming(cv::compile_args(kernels, networks));
+    } catch (const std::exception& ex) {
+        std::cerr << "Exception occured during pipeline construction: " << ex.what() << std::endl;
+        return -1;
+    }
+    // The execution part
+
+    // TODO USE may set pool size from outside and set queue_capacity size,
+    // compile arg: cv::gapi::streaming::queue_capacity
+    pipeline.setSource(std::move(cap));
+    pipeline.start();
+
+    int framesCount = 0;
+    cv::TickMeter t;
+    cv::VideoWriter writer;
+    cv::Mat outMat;
+    t.start();
+    while (pipeline.pull(cv::gout(outMat))) {
+        cv::imshow("Out", outMat);
+        cv::waitKey(1);
+        if (!output.empty()) {
+            if (!writer.isOpened()) {
+                const auto sz = cv::Size{outMat.cols, outMat.rows};
+                writer.open(output, cv::VideoWriter::fourcc('M','J','P','G'), 25.0, sz);
+                CV_Assert(writer.isOpened());
+            }
```

Comment:
It makes sense to open writer before the main processing loop.

Reply:
applied